catena4450m102_waterlevel/catena4450m102_waterlevel.ino

```diff
@@ -0,0 +1,697 @@
+                gCatena.SafePrintf(
+                        "Platform Flags:  %#010x\n",
+                        flags
+                        );
+                gCatena.SafePrintf(
+                        "Operating Flags:  %#010x\n",
+                        gCatena.GetOperatingFlags()
+                        );
+                }
+        else
+                {
+                gCatena.SafePrintf("**** no platform, check provisioning ****\n");
+                flags = 0;
+                }
+
+
+        /* initialize the lux sensor */
+        if (flags & CatenaBase::fHasLuxRohm)
+                {
+                bh1750.begin();
+                fLux = true;
+                bh1750.configure(BH1750_CONTINUOUS_HIGH_RES_MODE_2);
+                }
+        else
+                {
+                fLux = false;
+                }
+
+        /* initialize the BME280 */
+        if (!bme.begin(BME280_ADDRESS, Adafruit_BME280::OPERATING_MODE::Sleep))
+                {
+                gCatena.SafePrintf("No BME280 found: check wiring\n");
+                fBme = false;
+                }
+        else
+                {
+                fBme = true;
+                }
+
+        /* is it modded? */
+        uint32_t modnumber = gCatena.PlatformFlags_GetModNumber(flags);
+
+        fHasPower1 = false;
+
+        if (modnumber != 0)
+                {
+                gCatena.SafePrintf("Catena 4450-M%u\n", modnumber);
+                if (modnumber == 101)
+                        {
+                        fHasPower1 = true;
+                        kPinPower1P1 = A0;
+                        kPinPower1P2 = A1;
+                        }
+                else if (modnumber == 102 || modnumber == 103 || modnumber == 104)
+                        {
+                        fHasWaterTemp = flags & CatenaBase::fHasWaterOneWire;
+                        fSoilSensor = flags & CatenaBase::fHasSoilProbe;
+                        }
+                else
+                        {
+                        gCatena.SafePrintf("unknown mod number %d\n", modnumber);
+                        }
+                }
+        else
+                {
+                gCatena.SafePrintf("No mods detected\n");
+                fHasWaterTemp = flags & CatenaBase::fHasWaterOneWire;
+                fSoilSensor = flags & CatenaBase::fHasSoilProbe;
+                }
+
+        if (fHasWaterTemp)
+                {
+                sensor_WaterTemp.begin();
+                if (sensor_WaterTemp.getDeviceCount() == 0)
+                        {
+                        gCatena.SafePrintf("No one-wire temperature sensor detected\n");
+                        fFoundWaterTemp = false;
+                        }
+                else
+                        fFoundWaterTemp = true;
+                }
+
+        /* now, we kick off things by sending our first message */
+        gLed.Set(LedPattern::Joining);
+
+        // unit testing for the scaling functions
```

Comment:
Delete these comments (line 361 though 377)

Reply:
Sure, Thank you